src/vs/base/browser/ui/tree/abstractTree.ts

```diff
@@ -788,12 +798,30 @@ class FindWidget<T, TFilterData> extends Disposable {
 				right = this.right === 0 ? Number.POSITIVE_INFINITY : 0;
 			}
 
+			if (e.keyCode === KeyCode.UpArrow) {
+				top = 0;
+			} else if (e.keyCode === KeyCode.DownArrow) {
+				top = Number.POSITIVE_INFINITY;
+			}
+
 			if (right !== undefined) {
 				e.preventDefault();
 				e.stopPropagation();
 				this.right = right;
 				this.layout();
 			}
+
+			if (top !== undefined) {
+				e.preventDefault();
+				e.stopPropagation();
+				this.top = top;
+				const transition = this.elements.root.style.transition;
+				this.elements.root.style.transition = 'unset';
+				this.layout();
+				setTimeout(() => {
+					this.elements.root.style.transition = transition;
+				}, 100);
```

Comment:
Would [`transitionend`](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event) be more appropriate here?

Reply:
No, since there's no transition happening. But we can certainly use 0 for this.

```suggestion
				}, 0);
```